apps/faucet/src/endpoints/faucet.service.ts

```diff
@@ -0,0 +1,221 @@
+
+    return true;
+  }
+
+  private getTransaction(address: string, faucetToken: string, networkConfig: NetworkConfig): Transaction {
+    const tokenHex = this.padHex(Buffer.from(faucetToken.split('-').slice(0, 2).join('-'), 'ascii').toString('hex'));
+    const tokenAmountHex = this.padHex(BigInt(this.apiConfigService.config.faucetTokenAmount).toString(16));
+
+    const isNft = faucetToken.split('-').length === 3;
+    if (isNft) {
+      const tokenNonceHex = faucetToken.split('-')[2];
+      return new Transaction({
+        chainID: networkConfig.ChainID,
+        gasPrice: networkConfig.MinGasPrice,
+        gasLimit: 500000,
+        receiver: new Address(this.apiConfigService.config.faucetAddress),
+        data: new TransactionPayload(`ESDTNFTTransfer@${tokenHex}@${tokenNonceHex}@${tokenAmountHex}@${new Address(address).hex()}`),
+        sender: new Address(this.apiConfigService.config.faucetAddress),
+      });
+    }
+
+    return new Transaction({
+      chainID: networkConfig.ChainID,
+      gasPrice: networkConfig.MinGasPrice,
+      gasLimit: 500000,
+      receiver: new Address(address),
+      data: new TransactionPayload(`ESDTTransfer@${tokenHex}@${tokenAmountHex}`),
+      sender: new Address(this.apiConfigService.config.faucetAddress),
+    });
+  }
+
+  private async getNonce(): Promise<number> {
+    const value = await this.cachingService.get(CacheInfo.FaucetNonce.key);
+    if (!value && value !== 0) {
+      const accountNonce = await this.getLatestNonce();
+
+      await this.cachingService.set(CacheInfo.FaucetNonce.key, accountNonce, CacheInfo.FaucetNonce.ttl);
+
+      return accountNonce;
+    }
+
+    return await this.cachingService.incrementRemote(CacheInfo.FaucetNonce.key);
+  }
+
+  async getLatestNonce(): Promise<number> {
+    if (!this.faucetAccount) {
+      throw new Error('No faucet account initialized');
+    }
+
+    if (!this.provider) {
+      throw new Error('No provider initialized');
+    }
+
+    const account = await this.provider.getAccount(this.faucetAccount.address);
+    this.faucetAccount.update(account);
+
+    return this.faucetAccount.nonce.valueOf();
+  }
+
+  private padHex(value: string): string {
+    return (value.length % 2 ? '0' + value : value);
+  }
+
+  getSettings(): FaucetSettings {
+    const faucetToken = this.apiConfigService.config.faucetToken;
+
+    return {
+      address: this.apiConfigService.config.faucetAddress,
+      amount: this.apiConfigService.config.faucetAmount,
+      token: faucetToken ? faucetToken.split('-').slice(0, 2).join('-') : undefined,
+      tokenAmount: this.apiConfigService.config.faucetTokenAmount ? this.apiConfigService.config.faucetTokenAmount : undefined,
+      recaptchaBypass: this.shouldBypassCaptchaCheck(),
+    };
+  }
+
+  private async validateRecaptcha(recaptcha: string, clientIp: string | undefined): Promise<boolean> {
+    const result = await this.apiService.post('https://www.google.com/recaptcha/api/siteverify', qs.stringify({
```

Comment:
* maybe you can add the google api siteverify url in config file
* add try/catch blocks to ensure that if the request is failed to catch the error

Reply:
I will keep it here because we rely on their API and putting it into config can make someone think that it is customizable and can use other captcha services. added try/catch